src/command/supergraph/compose/do_compose.rs

```diff
@@ -260,31 +262,23 @@ impl Compose {
         }
 
         // Build the supergraph binary, paying special attention to the CLI options
-        let supergraph_binary = InstallSupergraph::builder()
-            .federation_version(fed_version.clone())
-            .elv2_license_accepter(self.opts.plugin_opts.elv2_license_accepter)
-            .studio_client_config(client_config.clone())
-            .and_override_install_path(override_install_path)
-            .skip_update(self.opts.plugin_opts.skip_update)
-            .build()
-            .install()
+        let supergraph_binary = InstallSupergraph::new(fed_version.clone(), client_config.clone())
+            .install(
+                override_install_path,
+                self.opts.plugin_opts.elv2_license_accepter,
+                self.opts.plugin_opts.skip_update,
+            )
             .await?;
 
-        // Federation versions and supergraph versions are synonymous, but have different types
-        // representing them depending on their use (eg, we only ever have an exact
-        // SupergraphVersion because we can only ever use an exact version of the binary, where the
-        // FederationVersion can just point to latest to get the latest version--these are similar,
-        // but represent different ways of using the underlying version)
-        let supergraph_version: SupergraphVersion = fed_version.clone().try_into()?;
```

Comment:
do you know for sure that using `fed_version.clone()` above is going to get the _right_ version from the supergraph binary's perspective?